### PR TITLE
[15.0][IMP] base_bank_from_iban: automatically create banks with schwifty

### PR DESCRIPTION
**Pull request by carlosdauden**
Backport from https://github.com/OCA/community-data-files/pull/170

@Tecnativa

### CODE REVIEW

**Comment by zamberjo** [comment]
Hi, maybe you might want to take a look at this https://github.com/OCA/community-data-files/pull/198

**Comment by carlosdauden** [comment]
@zamberjo version fixed

**Review by CarlosRoca13 — APPROVED** [review]

File: base_bank_from_iban/models/res_partner_bank.py [hunk 1]
```diff
@@ -9,10 +13,51 @@
...
+        self._add_bank_vals(vals)
+        return super().write(vals)
+
+    def _add_bank_vals(self, vals):
+        if vals.get("acc_number") and not vals.get("bank_id"):
+            try:
+                bank = self._get_bank_from_iban(vals["acc_number"])
+                vals["bank_id"] = bank.id
+            except Exception:
+                _logger.info("Could not find bank from IBAN", exc_info=True)
```
Comment (zamberjo) [inline base_bank_from_iban/models/res_partner_bank.py:37]: how do you see controlling a savepoint this part? to prevent exceptions + closed cursors.

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge major

**Pull request merged**